app/models/reference/CountryCode.scala

```diff
@@ -17,12 +17,23 @@
 package models.reference
 
 import cats.Order
-import play.api.libs.json._
+import config.FrontendAppConfig
+import play.api.libs.json.*
 
 case class CountryCode(code: String)
 
 object CountryCode {
 
+  def reads(config: FrontendAppConfig): Reads[CountryCode] =
```

Comment:
We can simplify the reads and writes here to something like

```
implicit val format: Format[CountryCode] = Json.valueFormat[CountryCode]

def reads(config: FrontendAppConfig): Reads[CountryCode] = {
  val key = if (config.isPhase6Enabled) "key" else "code"
  (__ \ key).read[String].map(CountryCode(_))
}
```

Think it's good to follow a familiar pattern to the other reference data models if we can.

Reply:
Sorted @jackheslop96